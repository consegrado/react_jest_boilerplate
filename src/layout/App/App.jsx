// @flow

import React from 'react';
import TabsDemo from '@/components/TabsDemo';

const App = () => {
  return (
    <>
      <TabsDemo />
    </>
  );
};

export default App;
