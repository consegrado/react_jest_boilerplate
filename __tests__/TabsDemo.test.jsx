// @flow

import React from 'react';
import App from '@/layout/App';
import { mount } from 'enzyme';

describe('App', () => {
  it('render', () => {
    const wrapper = mount(<App />);
    const tabLast = wrapper.find('[data-test-name="tab"]').last();
    tabLast.simulate('click');
    const tabContentLast = wrapper.find('[data-test-name="tab-content"]').last();

    expect(tabContentLast).toMatchSnapshot();
  });
});

//
// describe('check the possibility of changing tabs', () => {
//   it('check for lack of content tab2 before changing', () => {
//     const wrapper = mount(<TabsDemo />);
//     const tabContets = wrapper.find('[role="tabpanel"]');
//     expect(tabContets.filterWhere(n => n.text() === 'Any content 2')).toHaveLength(0);
//   });
//
//   it('check for the presence of content tab2 after changing', () => {
//     const wrapper = mount(<TabsDemo />);
//     const tabTitles = wrapper.find('[role="tab"]');
//     const tabContets = wrapper.find('[role="tabpanel"]');
//     tabTitles.filterWhere(n => n.text() === 'Title 2').simulate('click');
//     expect(tabContets.filterWhere(n => n.text() === 'Any content 2')).toHaveLength(1);
//   });
// });
