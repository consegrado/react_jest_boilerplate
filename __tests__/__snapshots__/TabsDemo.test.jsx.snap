// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render 1`] = `
<div
  data-test-name="tab-content"
>
  <h2>
    Any content 2
  </h2>
</div>
`;
